### ISSUE TITLE
Unexpected behavior with option: scrollDefault: 'now'

### ISSUE THREAD

**Issue by itslittlejohn** [jonthornton/jquery-timepicker#517]
Great plugin, but I have discovered what I think is a bug.

When the option `scrollDefault: 'now'` is set, conversion from military time seems to break.

See this fiddle to reproduce it. https://jsfiddle.net/n4coaygj/

Typing 0945 should turn into 9:45 AM but instead it is PM. Removing the `scrollDefault` option fixes the issue.

**Comment by jonthornton** [comment]
The time parser in timepicker doesn't know that a leading zero means 24hr time. I decided not to bother implementing 24hr input detection because it fails in the case of 10:00, 11:00, and 12:00 - it would be a partial solution at best.

You may notice that `scrollDefault: 'now'` only causes this problem in the evening. There's some logic to attempt to guess am/pm if that info is omitted: https://github.com/jonthornton/jquery-timepicker/blob/master/jquery.timepicker.js#L1146

I agree it's kinda odd behavior. Any ideas on a clean fix for it?

**Comment by itslittlejohn** [comment]
Ah! Somehow all along I've been under the impression that 24 hour time parsing was supported. I'm actually not sure why I specified the leading zero, because it makes sense to me that just typing `945` by itself should always be interpreted as AM. But even typing that results in `9:45 PM` in the afternoons as you noted.

Maybe I'm not understanding, but how does 24 hr detection fail for the scenarios you listed? It seems to work fine with this similar [jquery plugin](http://timepicker.co/options/). You can see how the algorithm works [here](https://github.com/wvega/timepicker/blob/master/jquery.timepicker.js#L736).

I would assume that typing `10` or `1000` would always result in `10:00 AM`. Entering `22` or `2200` would always convert to `10:00 PM`. Does that make sense?

**Comment by jonthornton** [comment]
tl;dr: use a 24 hour time format or set `scrollDefault: false` if this bugs you.

This behavior only happens when you have `scrollDefault` enabled. That setting scrolls the dropdown to a particular time, typically on the assumption that the user wants a time near the `scrollDefault` - it's a tool to help app developers provide sensible defaults. The typeahead will attempt to make intelligent guesses about am/pm based on this `scrollDefault`.

> 24 hour parsing should be doable?

If you've specified a 24 hour display format or have scrollDefault disabled, that's exactly what timepicker will do.

> how does 24 hr detection fail for the scenarios you listed?

If a time has a leading zero or an hour greater than 12, you can assume the input is 24 hour time format. An hour of 10, 11, or 12 is valid input for both 12 and 24 hour format, so you can't assume which the user intends.
